src/transactions/test/OfferTests.cpp

```diff
@@ -674,350 +674,435 @@ TEST_CASE("create offer", "[tx][offers]")
+                }
 
-                    // Offers are: sell 100 IDR for 150 USD; sell IRD @ 0.66
-                    // -> buy USD @ 1.5
-                    // first 6 offers get taken for 6*150=900 USD, gets 600
-                    // IDR in return
+                SECTION("offer crosses own")
+                {
+                    for_all_versions(*app, [&] {
+                        issuer.pay(a1, usd, 20000);
 
-                    // For versions < 10:
-                    // offer #7 : has 110 USD available
-                    //    -> can claim partial offer 100*110/150 = 73.333 ;
-                    //    -> 26.66666 left
-                    // 8 .. untouched
-                    // the USDs were sold at the (better) rate found in the
-                    // original offers
+                        // ensure we could receive proceeds from the offer
+                        a1.pay(issuer, idr, 50000);
 
-                    // offer is sell 1010 USD for 505 IDR; sell USD @ 0.5
-                    market.requireChangesWithOffer(
-                        {{offers[0].key, OfferState::DELETED},
-                         {offers[1].key, OfferState::DELETED},
-                         {offers[2].key, OfferState::DELETED},
-                         {offers[3].key, OfferState::DELETED},
-                         {offers[4].key, OfferState::DELETED},
-                         {offers[5].key, OfferState::DELETED},
-                         {offers[6].key, {idr, usd, price, 28}}},
-                        [&] {
-                            return market.addOffer(
-                                b1, {usd, idr, Price{1, 2}, 1009},
-                                OfferState::DELETED);
-                        });
+                        // offer is sell 150 USD for 100 IDR; sell USD @ 1.5 /
+                        // buy IRD @ 0.66
+                        REQUIRE_THROWS_AS(
+                            market.requireChangesWithOffer(
+                                {},
+                                [&] {
+                                    return market.addOffer(
+                                        a1, {usd, idr, exactCrossPrice, 150});
+                                }),
+                            ex_MANAGE_SELL_OFFER_CROSS_SELF);
+                    });
+                }
 
-                    market.requireBalances({{a1, {{usd, 1009}, {idr, 99328}}},
-                                            {b1, {{usd, 18991}, {idr, 672}}}});
-                });
+                SECTION("offer crosses and removes first")
+                {
+                    for_all_versions(*app, [&] {
+                        a1CheckSetAuthToMaintainLiabilities();
+                        issuer.pay(b1, usd, 20000);
 
-                for_versions_from(10, *app, [&] {
-                    issuer.pay(b1, usd, 20000);
+                        // offer is sell 150 USD for 100 USD; sell USD @ 1.5 /
+                        // buy IRD @ 0.66
```

Comment:
Nitpick: typo "IRD" -> "IDR" (this typo happens several times in this file)

Reply:
Fixed